src/client/app/helpers/custom.functions.helper.ts

```diff
@@ -1,30 +1,22 @@
 import { storage, environment, post, trustedSnippetManager } from './index';
 import { getDisplayLanguage } from '../strings';
 import { uniqBy } from 'lodash';
-// import {
-//     ensureFreshLocalStorage
-// } from '../helpers';
+import {
+    ensureFreshLocalStorage
+} from '../helpers';
+
+const isCustomFunctionRegex = /@customfunction/i;
 
 export function navigateToRegisterCustomFunctions() {
-    let allSnippetsToRegisterWithPossibleDuplicate: ICustomFunctionsRegistrationRelevantData[] =
+    ensureFreshLocalStorage();
+    let allSnippetsToRegisterWithPossibleDuplicate: ISnippet[] =
         uniqBy([storage.current.lastOpened].concat(storage.snippets.values()), 'id')
             .filter(snippet => trustedSnippetManager.isSnippetTrusted(snippet.id, snippet.gist, snippet.gistOwnerId))
-            .filter(snippet => snippet.customFunctions && snippet.customFunctions.content && snippet.customFunctions.content.trim().length > 0)
-            .map((snippet): ICustomFunctionsRegistrationRelevantData => {
-                try {
-                    return {
-                        name: snippet.name,
-                        data: JSON.parse(snippet.customFunctions.content)
-                    };
-                }
-                catch {
-                    throw new Error(`Error parsing metadata for snippet "${snippet.name}"`);
-                }
-            });
+            .filter(snippet => snippet.script && isCustomFunctionRegex.test(snippet.script.content));
 
     let data: IRegisterCustomFunctionsPostData = {
         snippets: allSnippetsToRegisterWithPossibleDuplicate,
-        displayLanguage: getDisplayLanguage()
+        displayLanguage: getDisplayLanguage(),
```

Comment:
For my understanding, why is the comma needed for the last property?

Reply:
If every line handles its own comma, then to delete a property, you can just delete the line. To add a property, you can just copy the line and modify the keys and values. 

I think (this)[https://medium.com/@nikgraf/why-you-should-enforce-dangling-commas-for-multiline-statements-d034c98e36f8] summarizes it better.

Reply:
I also just like the uniform-ness of it ;P

Reply:
Sounds good to me!